service/init.go

```diff
@@ -317,32 +344,92 @@ func (pcf *PCF) Terminate() {
 	logger.InitLog.Infof("PCF terminated")
 }
 
-func (pcf *PCF) registerNF() {
+func (pcf *PCF) BuildAndSendRegisterNFInstance() (models.NfProfile, error) {
+	self := context.PCF_Self()
+	profile, err := consumer.BuildNFInstance(self)
+	if err != nil {
+		initLog.Error("Build PCF Profile Error: %v", err)
+	}
+	initLog.Infof("Pcf Profile Registering to NRF: %v", profile)
+	//Indefinite attempt to register until success
+	profile, _, self.NfId, err = consumer.SendRegisterNFInstance(self.NrfUri, self.NfId, profile)
+	return profile, err
+}
 
+func (pcf *PCF) RegisterNF() {
 	for {
+		msg := <-ConfigPodTrigger
 		//wait till Config pod updates config
-		if msg := <-ConfigPodTrigger; msg {
+		if msg {
 			initLog.Infof("Config update trigger %v received in PCF App", msg)
-			self := context.PCF_Self()
-			profile, err := consumer.BuildNFInstance(self)
-			if err != nil {
-				initLog.Error("Build PCF Profile Error: %v", err)
-			}
-			initLog.Infof("Pcf Profile Registering to NRF: %v", profile)
-			//Indefinite attempt to register until success
-			_, self.NfId, err = consumer.SendRegisterNFInstance(self.NrfUri, self.NfId, profile)
+			profile, err := pcf.BuildAndSendRegisterNFInstance()
 			if err != nil {
 				initLog.Errorf("PCF register to NRF Error[%s]", err.Error())
 			} else {
+				//stop keepAliveTimer if its running
+				pcf.StopKeepAliveTimer()
```

Comment:
move pcf.StopKeepAliveTimer() to pcf.StartKeepAliveTimer(profile) as this confuses why stop timer and then immediately start again. Or probably it should be named as updateKeepAliveTimer()